tools/wiki_sync.py

```diff
@@ -0,0 +1,173 @@
+    # Uses helper functions to find .md files and paths, to parse the Github Wiki formatting spec
+    # Creates a map of the formatting spec to document information
+    # Map is used to create a sidebar file, and a copy paste the files from the Pocket Repository to Github Wiki
+
+    sidebar = defaultdict(list)
+    for path in get_all_markdown_file_paths():
+        wiki_path_format = get_file_to_wiki_comment(path)
+
+        if not wiki_path_format:
+            continue
+
+        dirname = os.path.dirname(wiki_path_format)
+        file = os.path.basename(wiki_path_format)
+        categories = dirname.split("/")
+        display_name = " ".join([s.title() for s in file.split("_")])
+        filename = f"{categories[-1].title()} {display_name.title()}"
+
+        sidebar[dirname].append(DocumentationFile(filename, display_name, path))
+
+    return sidebar
+
+
+def write_sidebar_file(sidebar: Dict[str, List[DocumentationFile]]) -> None:
+    sidebar_format = "'''Contents'''\n"
+    sidebar_format += "*'''[[ Home | Home ]]'''\n"
+
+    depth = 1  # helps track the level of nesting in the wiki sidebar
+    for category, doc_files in sidebar.items():
+        if category == "home":
+            continue
+
+        subcategories = category.split("/")
+        for subcategory in subcategories:
+            sidebar_format += ("*" * depth) + f"'''{subcategory.title()}'''\n"
+            depth += 1
+
+        for doc in doc_files:
+            sidebar_format += (
+                "*" * depth
+            ) + f"[[ {doc.file_name} | {doc.display_name} ]]\n"
+
+        # reset depth for the next category
+        depth = 1
+
+    with open(f"{WIKI_DIR}/_Sidebar.mediawiki", "w") as f:
+        f.write(sidebar_format)
+
+
+def write_wiki_pages(sidebar: Dict[str, List[DocumentationFile]]) -> None:
+    # open md files in the repo an prepare for migration process
+    
+    for category, doc_files in sidebar.items():
+        for doc_file in doc_files:
+            with open(doc_file.path) as source:
+                target = f"{WIKI_DIR}/{doc_file.file_name}.md"
+                if category == "home":
+                    target = f"{WIKI_DIR}/Home.md"
+
+                with open(target, "w") as dest:
+                    dest.write(source.read())
+
+
+def run_wiki_migration():
+    os.makedirs(TEMP_WIKI, exist_ok=True)
+
+    # repo env variables
+    secret = os.environ["USER_TOKEN"]
+    user_name = os.environ["USER_NAME"]
+    user_email = os.environ["USER_EMAIL"]
+    owner = os.environ["OWNER"]
+    repo_name = os.environ["REPOSITORY_NAME"]
+
+    # init, pull, delete
+    subprocess.call(["git", "init"], cwd=TEMP_WIKI)
+    subprocess.call(["git", "config", "user.name", f"{user_name}"], cwd=TEMP_WIKI)
+    subprocess.call(["git", "config", "user.email", f"{user_email}"], cwd=TEMP_WIKI)
+    subprocess.call(
+        ["git", "pull", f"https://{secret}@github.com/{owner}/{repo_name}.wiki.git"],
```

Comment:
It doesn't appear we can create a personal access token under organization, unless it is a [fine-grained PAT](https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/). I don't want us to utilize tokens that belong to particular people, and would rather use ones that are tied to `pokt-network` organization. The problem with fine-grained PAR, however, is it is not obvious [which permissions](https://docs.github.com/en/rest/overview/permissions-required-for-fine-grained-personal-access-tokens?apiVersion=2022-11-28) to add to allow to push to wiki. I am happy to research and figure out which permissions are actually needed, BUT I suspect there is a better way. Let's try this first, and if it doesn't work we can defer to fine-grained PAT.

From inside the GitHub action CI we should have all necessary permissions to interact with wiki that belongs to the same repo with `GITHUB_ACTOR` and `GITHUB_TOKEN` variables. While `GITHUB_ACTOR` environment variable is available in runtime, `GITHUB_TOKEN` is a secret that should be passed as an environment variable. I'll add a similar suggestion to `.github/workflows/wiki_sync_process.yml` to add it.

Feel free to move `os.environ` to separate variables, I just want to demonstrate how this pull CLI should look like:

```suggestion
        ["git", "pull", f"https://{os.environ["GITHUB_ACTOR"]}:{os.environ["GITHUB_TOKEN"]}@github.com/{owner}/{repo_name}.wiki.git"],
```

Reply:
@okdas Appreciate the suggestions and explanation!

Wanted to ask: so should this just "work out of the box" if we make these changes or do we need to set up additional secrets in the repo settings?

Reply:
> Wanted to ask: so should this just "work out of the box" if we make these changes or do we need to set up additional secrets in the repo settings?

Yes, it should! This is my understanding from a brief research – that `GITHUB_TOKEN`, which is a unique secret generated for each individual job run, is supposed to have access to pull/push wiki data for the same repo. This would be ideal, of course, so we don't have to manage additional secrets associated with people.

Reply:
Ah interesting. This seems to have worked when I did it on my own fork. 
For reference if it needs to be reviewed: I used "Generate New Token (Classic)" and just gave it the whole "Repo" access - maybe it can be even more specific than that. 

Lmk how it goes.